core/src/avm2/e4x.rs

```diff
@@ -278,7 +300,10 @@ impl<'gc> E4XNode<'gc> {
             let is_whitespace_char = |c: &u8| matches!(*c, b'\t' | b'\n' | b'\r' | b' ');
             let is_whitespace_text = text.iter().all(is_whitespace_char);
             if !(text.is_empty() || ignore_white && is_whitespace_text) {
-                let text = AvmString::new_utf8_bytes(activation.context.gc_context, text);
+                let text = AvmString::new_utf8_bytes(
+                    activation.context.gc_context,
+                    if is_text { trim_ascii(text) } else { text },
```

Comment:
I think needs to be `if is_text && ignore_white {` so that it wouldn't trim if `XML.ignoreWhitespace` is false. Although we don't have that hooked up yet (line 245 hardcodes it to true at the moment)